### PR TITLE
Montana Capital Gains Credit

### LINKED ISSUE #2980: Montana Capital Gains Credit
Montana Capital Gains Credit

https://rules.mt.gov/gateway/RuleNo.asp?RN=42%2E4%2E502

### PR DESCRIPTION
**Pull request by Kiki-67**
Fixes #2980

### Summary
📱🛠️🌐

Add a new policy parameter `ubi_amount` to the US tax and benefit system. Implement a basic universal basic income (UBI) reform that replaces the standard deduction and personal exemption with a flat monthly payment. Update the `reform_us.yaml` file and the `policyengine-us.ipynb` notebook to demonstrate the UBI reform.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Add a new parameter `ubi_amount` to the UBI reform function ()

### CODE REVIEW

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gains/percent.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+description: Montana allows filers to deduct this percentage of their net capital gains.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gains/percent.yaml:1]:
```suggestion
description: Montana allows a credit of this percentage of the filers net capital gains.
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/capital_gains.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/capital_gains.py:1]: rename file to mt_capital_gain_credit.py

File: policyengine_us/tests/policy/baseline/gov/states/mt/tax/income/credits/capital_gains.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/mt/tax/income/credits/capital_gains.yaml:1]: rename file to mt_capital_gain_credit.yaml

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gains/percent.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gains/percent.yaml:1]: rename folder to capital_gain

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gains/percent.yaml:1]: rename file to percentage.yaml

File: policyengine_us/variables/gov/states/mt/tax/income/credits/capital_gains.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    definition_period = YEAR
+    reference = "https://rules.mt.gov/gateway/RuleNo.asp?RN=42%2E4%2E502"
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.mt.tax.income.credits.capital_gains
+
+        net_capital_gain = tax_unit("net_capital_gain", period)
+
+        return p.percent * net_capital_gain
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/capital_gains.py:18]:
```suggestion
        return p.percentage * net_capital_gain
```

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Montana capital gain credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Montana net capital gain credit.
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gain/percentage.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+description: Montana allows a credit of this percentage of the filers net capital gains.
+values:
+  2021-01-01: 0.02
+metadata:
+  unit: /1
+  label: Montana capital gain credit percentage
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gain/percentage.yaml:6]:
```suggestion
  label: Montana capital gains credit rate
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gain/percentage.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+description: Montana allows a credit of this percentage of the filers net capital gains.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/capital_gain/percentage.yaml:1]:
```suggestion
description: Montana provides a credit for this fraction of the filer's net capital gains.
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/mt_capital_gain_credit.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    label = "Montana capital gain credit"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://rules.mt.gov/gateway/RuleNo.asp?RN=42%2E4%2E502"
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.mt.tax.income.credits.capital_gain
+
+        net_capital_gain = tax_unit("net_capital_gain", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/mt_capital_gain_credit.py:16]: Should this be a negative credit if they have capital losses? Please add a test for this and cap at zero if needed. Or if the capping occurs in net_capital_gain (I'm on my phone so can't check) please add a comment noting that.

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Please also add it to a list of refundable or non refundable credits

File: policyengine_us/variables/gov/states/mt/tax/income/credits/mt_capital_gain_credit.py (hunk 6 above)
Comment (Kiki-67) [reply policyengine_us/variables/gov/states/mt/tax/income/credits/mt_capital_gain_credit.py:16]: I will leave a comment # The net capital gain variable is capped at 0

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Please add it to the list of nonrefundable/refundable credits

**Review by MaxGhenis — APPROVED** [review]
Just a note that Montana calls this the Capital Gain Credit, rather than the typical Gains nomenclature

**Pull request merged**